pyscript.core/docs/README.md

```diff
@@ -0,0 +1,341 @@
+
+```html
+<script type="micropython">
+    # XWorker is globally defined
+    print(XWorker != None)
+</script>
+```
+
+Please read the [XWorker](#xworker) dedicated section to know more.
+
+  </div>
+</details>
+
+
+## How Events Work
+
+Inspired by the current [HTML Standard](https://html.spec.whatwg.org/multipage/webappapis.html#event-handlers):
+
+> the event handler is exposed through a name, which is a string that always starts with "_on_" and is followed by the name of the event for which the handler is intended.
+
+We took a similar approach, replacing that `on` prefix with whatever *interpreter* or *custom type* is available on the page, plus a *dash* `-` to avoid clashing with standards:
+
+```html
+<script type="micropython">
+    def print_type(event, double):
+        # logs "click 4"
+        print(f"{event.type} {double(2)}")
+</script>
+<button micropython-click="print_type(event, lambda x: x * 2)">
+    print type
+</button>
+```
+
+If this example felt a bit verbose, be ensured custom types would work the same:
+
+```html
+<!-- ℹ️ - requires py-script custom type -->
+<button py-click="print(event.type)">
+    print type
+</button>
+```
+
+What is important to understand about *events* in PyScript is that the text within the attribute is executed just like any other inline or external content is, through the very same *interpreter*, with the notably extra feature that the `event` reference is made temporarily available as *global* by *core*.
+
+This really reflects how otherwise native Web inline events handlers work and we think it's a great feature to support ... *but*:
+
+ * if your script runs *asynchronously* the `event` might be gone on the main / UI thread and by that time any of its `event.stopPropagation()` or `event.preventDefault()` goodness will be problematic, as too late to be executed
+ * if your *interpreter* is *experimental*, or incapable of running *synchronous* events, the `event` reference might be less useful
+
+ℹ️ - Please note that if your code runs via *XWorker*, hence in a different thread, there are different caveats and constraints to consider. Please read the [XWorker](#xworker) dedicated section to know more.
+
+#### The type-env
+
+As `env` as attribute has been mentioned already, it is also possible to confine events to a specific *env* by using the *interpreter* or *custom type* name, and followed by `-env`, for the attribute value, providing the desired *env* to interpret and execute the listener.
+
+```html
+<script type="micropython">
+    def log():
+        print(1)
+</script>
+<!-- note the env value -->
+<script type="micropython" env="two">
+    # the button will log 2
+    def log():
+        print(2)
+</script>
+<!-- note the micropython-env value -->
+<button
+    micropython-env="two"
+    micropython-click="log()"
+>
+    log
+</button>
+```
+
+As mentioned before, this will work with `py-env` too, or any custom type defined out there.
+
```

Comment:
```suggestion

It is not currently possible for an element to have register event listeners for two different interpreters of the same type using this syntax.
```
I think this is true?

Reply:
not sure I am following ... any example you have in mind so I can add it to disambiguate / avoid confusion?

Reply:
For instance, there's no way to have 'click' responders associated to two different envs on the same element, right?

```html
<script type="micropython" env="one">
    def foo():
        print("FOO!")
</script>
<script type="micropython" env="two">
    def bar():
        print("BAR!")
</script>

<!-- 
    There's no way to link this button to functions in both env one and env two, right?
    I.e. Is there a way for a click to run both foo() and bar()? 
    Or for a click to run foo() and a mouseover to run bar()?
-->
<button micropython-env="???" micropython-click="???">Click Me</button>
```

Reply:
I think there's no use case for that so I am not sure this is a real-world concern ... *env* is an **advanced** feature and whoever uses it would realize it's never desired to have other envs meshed up with listeners as it completely defeat the purpose of an env ... you bootstrap two different worlds, why would you want to trigger listeners all over the place?